platform-hub-web/src/app/kubernetes-tokens/kubernetes-tokens-list.component.js

```diff
@@ -0,0 +1,100 @@
+export const KubernetesTokensListComponent = {
+  template: require('./kubernetes-tokens-list.html'),
+  controller: KubernetesTokensListController
+};
+
+function KubernetesTokensListController(roleCheckerService, hubApiService, logger, $stateParams, $mdDialog) {
+  'ngInject';
+
+  const ctrl = this;
+  const userId = $stateParams.userId;
+
+  ctrl.searchSelectedUser = null;
+  ctrl.loading = true;
+  ctrl.isAdmin = false;
+  ctrl.tokens = [];
+  ctrl.searchText = '';
+  ctrl.user = null;
+
+  ctrl.searchUsers = searchUsers;
+  ctrl.deleteToken = deleteToken;
+  ctrl.filterKubernetesTokensByUser = filterKubernetesTokensByUser;
+
+  init();
+
+  function init() {
+    loadAdminStatus();
+    if (userId) {
+      loadUser(userId);
+      loadKubernetesTokens(userId);
+    }
+  }
+
+  function filterKubernetesTokensByUser() {
+    if (!ctrl.isAdmin && !ctrl.isProjectManager) {
+      return;
+    }
+    loadUser(ctrl.searchSelectedUser.id);
+    loadKubernetesTokens(ctrl.searchSelectedUser.id);
```

Comment:
Both of these methods set `ctrl.loading` back to `false` when whichever one finishes first. Probably not a big deal (just means data flashes in). But a better approach is to wrap both API calls in a `$q.all` and reset the loading parameter on a `finally` handler on the overall promise. There should be an example of this elsewhere in the codebase, or I can help with this if needed.